spot_rl_experiments/spot_rl/envs/nav_env.py

```diff
@@ -207,7 +208,20 @@ def get_success(self, observations):
         return succ
 
     def get_observations(self):
-        return self.get_nav_observation(self.goal_xy, self.goal_heading)
+        # Update the point goal given depth and rgb images
+        if self._enable_nav_goal_change:
+            arm_depth, arm_depth_bbox = self.get_gripper_images()
+            # Update the object x, y here
+            obj_xy = self.spot.get_new_goal_given_obj_img(arm_depth, arm_depth_bbox)
+            # Update the obj_xy to consider the base offset
+            obj_nav_offset_x = -1.5
+            goal_x, goal_y, _ = self.spot.get_global_from_local_based_on_input_T(
+                obj_nav_offset_x, 0, 0, obj_xy[0], obj_xy[1], 0
```

Comment:
If this offset is in global frame, then adjustment along only x axis may not solve the issue. We may need a more generalizable approximation function.

@jimmytyyang  as we discussed

Reply:
@tusharsangam you might want to rewrite something here :)